From 1f85c55247e67045a392ae9ceee1e58841a12508 Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Thu, 7 May 2015 18:26:57 +0800
Subject: [PATCH] usb: android gadget: remove self-powered flag

From compliance requirement that "Peripherals that return "Self-powered"
in the GetStatus(DEVICE) call are prohibited from drawing more than 100mA
at any time.". So remove self-powered flag as MaxPower is 500mA.

Change-Id: I68eac1a233a387ab15b652957baceb6153645f93
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-33441
Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 drivers/usb/gadget/android.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 845207d2e74a..c4f42a1f7f8a 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -162,7 +162,7 @@ static struct usb_configuration android_config_driver = {
 	.label		= "android",
 	.unbind		= android_unbind_config,
 	.bConfigurationValue = 1,
-	.bmAttributes	= USB_CONFIG_ATT_ONE | USB_CONFIG_ATT_SELFPOWER,
+	.bmAttributes	= USB_CONFIG_ATT_ONE,
 	.MaxPower	= 500, /* 500ma */
 };
 
@@ -1552,7 +1552,6 @@ static int android_bind(struct usb_composite_dev *cdev)
 	device_desc.iSerialNumber = id;
 	dev->reset_string_id = id;
 
-	usb_gadget_set_selfpowered(gadget);
 	dev->cdev = cdev;
 
 	return 0;
-- 
1.9.1

